openstack/ces/v1/metricdata/requests.go

```diff
@@ -4,186 +4,243 @@ import (
-	// 查询数据截止时间UNIX时间戳，单位毫秒。from必须小于to。
-	To string `q:"to,required"`
+	var res []BatchMetricData
+	err = raw.ExtractIntoSlicePtr(&res, "metrics")
+
+	return res, err
 }
 
-type AddMetricDataItem struct {
-	// 指标数据。
+// ----------------------------------------------------------------------------
+
+type MetricDataItem struct {
+	// Specifies the metric data.
 	Metric MetricInfo `json:"metric" required:"true"`
-	// 数据的有效期，超出该有效期则自动删除该数据，单位秒，最大值604800。
+	// Specifies the data validity period.
+	// The unit is second. The value range is 0–604,800 seconds.
+	// If the validity period expires, the data will be automatically deleted.
 	Ttl int `json:"ttl" required:"true"`
-	// 数据收集时间  UNIX时间戳，单位毫秒。  说明： 因为客户端到服务器端有延时，因此插入数据的时间戳应该在[当前时间-3天+20秒，当前时间+10分钟-20秒]区间内，保证到达服务器时不会因为传输时延造成数据不能插入数据库。
-	CollectTime int `json:"collect_time" required:"true"`
-	// 指标数据的值。
+	// Specifies when the data was collected.
+	// The time is UNIX timestamp (ms) format.
+	CollectTime int64 `json:"collect_time" required:"true"`
+	// Specifies the monitoring metric data to be added.
+	// The value can be an integer or a floating point number.
 	Value float64 `json:"value" required:"true"`
-	// 数据的单位。
+	// Specifies the data unit.
+	// Enter a maximum of 32 characters.
 	Unit string `json:"unit,omitempty"`
-	// 数据的类型，只能是\"int\"或\"float\"
+	// Specifies the enumerated type.
+	// Possible values:
+	// int
+	// float
 	Type string `json:"type,omitempty"`
 }
 
-// 指标信息
 type MetricInfo struct {
-	// 指标维度
+	// Specifies the metric dimension. A maximum of three dimensions are supported.
 	Dimensions []MetricsDimension `json:"dimensions" required:"true"`
-	// 指标名称，必须以字母开头，只能包含0-9/a-z/A-Z/_，长度最短为1，最大为64。  具体指标名请参见查询指标列表中查询出的指标名。
+	// Specifies the metric ID. For example, if the monitoring metric of an ECS is CPU usage, metric_name is cpu_util.
 	MetricName string `json:"metric_name" required:"true"`
-	// 指标命名空间，，例如弹性云服务器命名空间。格式为service.item；service和item必须是字符串，必须以字母开头，只能包含0-9/a-z/A-Z/_，总长度最短为3，最大为32。说明： 当alarm_type为（EVENT.SYS| EVENT.CUSTOM）时允许为空。
+	// Query the namespace of a service.
 	Namespace string `json:"namespace" required:"true"`
 }
 
-// 指标维度
 type MetricsDimension struct {
-	// 维度名
+	// Specifies the dimension. For example, the ECS dimension is instance_id.
+	// For details about the dimension of each service, see the key column in Services Interconnected with Cloud Eye.
+	// Start with a letter. Enter 1 to 32 characters.
+	// Only letters, digits, underscores (_), and hyphens (-) are allowed.
 	Name string `json:"name,omitempty"`
-	// 维度值
+	// Specifies the dimension value, for example, an ECS ID.
+	// Start with a letter or a digit. Enter 1 to 256 characters. Only letters, digits, underscores (_), and hyphens (-) are allowed.
 	Value string `json:"value,omitempty"`
 }
 
-func (opts AddMetricDataItem) ToMap() (map[string]interface{}, error) {
-	return golangsdk.BuildRequestBody(opts, "")
-}
+func CreateMetricData(client *golangsdk.ServiceClient, req []MetricDataItem) error {
```

Comment:
I think we decided to use opts instead or rec, or no?

Reply:
Although I think either req or opts are similar, let's use opts.